### ISSUE TITLE
Add new gateway proxy 

### ISSUE THREAD

**Issue by derrickburns** [solo-io/gloo#1785]
There is a bias toward creating all gateway proxy deployments/pods when gloo is installed. Yes, one can create the Kubernetes resources manually.  However, the pod template is now rather complex.  

It would be helpful to enable installing a single gateway proxy using the helm chart or glooctl without disturbing the other resources.  

Consider this. One can install a new application in a new namespace, but cannot connect it to the internet via gloo gateway without installing gloo again.

**Comment by ilackarms** [comment]
is it a reinstallation if none of the gloo manifests changed except the addition of a gateway proxy?

**Comment by derrickburns** [comment]
Nope

**Comment by derrickburns** [comment]
@ilackarms 
The connection between the _Gateway custom resource_ and the **gateway proxy** is rather difficult to manage.  Here is my situation.

That _Gateway CR_ explicitly names the gateway proxies that it impacts.  The **gateway proxies** must be created via a Kubernetes standard resources (i.e. Deployments).  However, there is no easy way to declare a Gateway CR in a namespace and its accompanying Deployment in the same namespace without also installing all the other control plane resources in said namespaces using `glooctl` or `helm`. 

Moreover, because of the complexity of the Helm (2) installation process, using `helm` directly is challenging for upgrades.  

Consider this use case:  I add a new service with its own virtual services.  

Now, the new virtual services need to be associated with a Gateway.  If I don' provide a Gateway CR, then either the new service must run in a namespace that already has a Gloo control plane, or I share a global Gloo control plane.  If the latter, then the Gateway must be configured to find the new virtual services.  And, if so, then the new service must use the Gateway in a compatible way (e.g. proxy proto, access logging, etc.) . If not, then the new service must define its own Gateway CR and its own Gloo control plane. 

In any case, I need to modify the *annotations* of the K8s Service to the Gateway proxies to add the hostnames so that `external-dns` will manage my DNS aliases.  This means that I cannot deploy a new set of virtual services without deploying a new Gloo control plane OR modifying a K8s Service outside the namespace of the new virtual services.

I cannot deploy a new Gloo control plane using GitOps because my Gitops provider (Flux) only supports Helm 2 at present, and the new Gloo helm chart is not backward compatible.

So that means that adding a new namespace/set of virtual services means that I must also modify a global K8s Service.

In other words, it is still pretty darn hard to offer self-service on a per-namespace basis with GitOps. 

**What would be better would be for the Gateway control plane service to create new K8s Deployments for Gateway proxies** and to remove the specification of any Gateway proxies from the helm chart.

**Comment by rickducott** [comment]
I'm merging this with https://github.com/solo-io/gloo/issues/2295

**Comment by rickducott** [comment]
From @ilackarms on the other ticket:

> the high number of fields that must line up between the kube svc and the proxy CRD for an instance of Envoy makes user workflows for configuring routing to their proxies difficult, e.g. in situations when configuring an external load balancer to the proxy requires setting health checks to the right listener
> gloo (or a gloo operator) should assume control of this behavior; i.e. that the helm chart is no longer responsible for creating *-proxy services, but that is left to a controller in Gloo itself.
> the same could be (but is not necessarily) applied to proxy Deployment/DaemonSets as well
